File: ebook-backend/src/main/java/com/ebookreader/ebook_backend/modules/subscription/entity/Plan.java
package com.ebookreader.ebook_backend.modules.subscription.entity;

import com.ebookreader.ebook_backend.common.base.BaseEntity;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

import java.math.BigDecimal;

@Entity
@Table(name = "plans")
@SQLDelete(sql = "UPDATE plans SET is_deleted = true WHERE id=?")
@SQLRestriction("is_deleted = false")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Plan extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Builder.Default
    private String currency = "TRY";

    @Builder.Default
    private int durationDays = 30;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SubscriptionPlan subscriptionPlan;

    @Builder.Default
    private boolean isActive = true;
}

File: ebook-frontend/src/features/subscription/types/plan.ts

export interface Plan {
    id: number;
    name: string;
    description: string;
    price: number;
    currency: string;
    durationDays: number;
    subscriptionPlan: 'FREE' | 'PREMIUM' | 'VIP';
    active: boolean;
}

export interface CreatePlanRequest {
    name: string;
    description?: string;
    price: number;
    durationDays: number;
    subscriptionPlan: 'FREE' | 'PREMIUM' | 'VIP';
}
